Pod/Classes/Internal/SEGSegmentIntegration.m

```diff
@@ -592,4 +607,52 @@ - (NSString *)getUserId
     return [[NSUserDefaults standardUserDefaults] valueForKey:SEGUserIdKey] ?: [[NSString alloc] initWithContentsOfURL:self.userIDURL encoding:NSUTF8StringEncoding error:NULL];
 }
 
+-(BOOL)hasPreviousVersion
+{
+    NSString *const SEGBuildKey = @"SEGBuildKey";
+    NSInteger previousBuild = [[NSUserDefaults standardUserDefaults] integerForKey:SEGBuildKey];
+    NSInteger currentBuild = [[[NSBundle mainBundle] infoDictionary][@"CFBundleShortVersionString"] integerValue];
+    
+    if (currentBuild > previousBuild) {
+        return true;
+    } else {
+        return false;
+    }
+}
+- (void)transferQueue
+{
+    NSArray *oldQueue = [self NSUserDefaultQueue];
+    // If there is a queue Array in NSUserDefaults
+    if (oldQueue) {
+        // Remove the old queue from NSUserDefaults and store on disk
+        [[NSUserDefaults standardUserDefaults] removeObjectForKey:SEGQueueKey];
+        SEGLog(@"[[NSUserDefaults standardUserDefaults] removeObjectForKey: %@]", SEGQueueKey);
+        [[oldQueue copy] writeToURL:[self queueURL] atomically:YES];
+        SEGLog(@"[[%@ copy] writeToURL:[%@] atomically:YES]", oldQueue, [self queueURL]);
+    }
+}
+
+- (void)transferTraits
+{
+    NSDictionary *oldTraits = [self NSUserDefaultTraits];
+    // If there is a traits dict in NSUserDefaults
+    if (oldTraits) {
+        // Remove the old traits
+        [[NSUserDefaults standardUserDefaults] removeObjectForKey:SEGTraitsKey];
+        SEGLog(@"[[NSUserDefaults standardUserDefaults] removeObjectForKey: %@]", SEGTraitsKey);
+        [[oldTraits copy] writeToURL:[self traitsURL] atomically:YES];
```

Comment:
Could we possibly be over-writing existing data there? Is it possible for previous version of the SDK to somehow have both stored data in `NSUserDefaults` as well as on disk as files?